### ISSUE TITLE
Maven download failure reason not clear

### ISSUE THREAD

**Issue by nfuller** [bazelbuild/bazel#16149]
### Description of the bug:

Please see: https://github.com/google/conscrypt/issues/1080

TL;DR - bazel wasn't able to automatically resolve / download an architecture-dependent jar and there were no errors reported that made it obvious what was going wrong. I've provided more info and a repro case linked to the conscrypt issue above.

The conscrypt team suggested the logging could be improved in bazel, and I agree. Also:

"PS if Bazel has an osdetector equivalent, we'd welcome a PR to update https://github.com/google/conscrypt/blob/master/README.md with the recipe!"

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

I've shared an example project here for easy repro: https://drive.google.com/file/d/1tdCAGkkTxCDZyZwku8ttYwoagZx2XOmG/view?usp=sharing - it should be accessible to Googlers.

Running bazel build java/example should demonstrate that Tink classes can be found, but a Conscrypt one cannot. It's also trivial to switch to the -uber version of the jar to confirm that works.

### Which operating system are you running Bazel on?

MacOS and Linux

### What is the output of `bazel info release`?

release 5.2.0

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

```text
N/A
```

### Have you found anything relevant by searching the web?

No

**Comment by Wyverald** [comment]
If I'm reading this correctly, this is a feature request for rules_jvm_external. Please file the issue in that repo (https://github.com/bazelbuild/rules_jvm_external).
